apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient5-plugin/src/main/java/co/elastic/apm/agent/httpclient/v5/helper/FutureCallbackWrapper.java

```diff
@@ -107,11 +107,10 @@ private void finishSpan(@Nullable Exception e) {
         // start by reading the volatile field
         final Span<?> localSpan = span;
         try {
-            if (context != null) {
-                Object responseObject = context.getAttribute(HttpCoreContext.HTTP_RESPONSE);
-                if (responseObject instanceof HttpResponse) {
-                    int statusCode = ((HttpResponse) responseObject).getCode();
-                    span.getContext().getHttp().withStatusCode(statusCode);
+            if (context instanceof HttpCoreContext) {
+                HttpResponse response = ((HttpCoreContext) context).getResponse();
+                if (response != null) {
+                    span.getContext().getHttp().withStatusCode(response.getCode());
```

Comment:
[for reviewer] this is an API change that is not strictly required, but allows to make it work with more recent versions of httpclient (tested locally with 5.4.4).